plugins/catalog-react/api-report.md

```diff
@@ -284,13 +284,26 @@ export type EntityListContextProps<
   queryParameters: Partial<Record<keyof EntityFilters, string | string[]>>;
   loading: boolean;
   error?: Error;
+  pageInfo?: {
+    next?: () => void;
+    prev?: () => void;
+  };
 };
 
 // @public
 export const EntityListProvider: <EntityFilters extends DefaultEntityFilters>(
-  props: PropsWithChildren<{}>,
+  props: EntityListProviderProps,
 ) => React_2.JSX.Element;
 
+// @public (undocumented)
+export type EntityListProviderProps = PropsWithChildren<{
+  enablePagination?:
+    | boolean
+    | {
+        limit?: number;
+      };
```

Comment:
Thinking that perhaps we should move the config reading into this one instead? Rather than having the catalog page read it and forward it. That way we don't add API surface here yet.

Not super opposed to this either, but then we'd need to consider the naming a bit more, and we might as well make the config flag `catalog.pagination` since this would make it stable API. Naming wise I'd think this prop should perhaps be just `paginated`?

```tsx
<EntityListProvider paginated>

<EntityListProvider paginated={{ limit: 20 }}>
```

Reply:
To summarize a discussion we had on the side: We're going with `<EntityListProvider pagination>` instead, since that works better for a future where we want to flip the default and allow `<EntityListProvider pagination={false}>`. We'll also drop the config altogether since that doesn't work for apps with a customized catalog page, and instead introduce it as props all the way to the `CatalogIndexPage`